apps/web/src/lib/functions/file-loaders/epub/generate-epub-html.ts

```diff
@@ -139,32 +139,41 @@ export default function generateEpubHtml(
       htmlHref = itemIdToHtmlRef[itemIdRef];
     }
 
-    const regexResult = /.*<body(?:[^>]*id="(?<id>.+?)")*[^>]*>(?<body>(.|\s)+)<\/body>.*/.exec(
-      data[htmlHref] as string
-    )!;
+    const contentParser = new DOMParser();
+    const parsedContent = contentParser.parseFromString(data[htmlHref] as string, 'text/html');
 
-    const bodyId = regexResult?.groups?.id || '';
-    let innerHtml = regexResult?.groups?.body || '';
+    const htmlClass = parsedContent.querySelector('html')?.className || '';
```

Comment:
would be nice to add some internal default class (e. g. ttu-book-html-wrapper) to identify this node for an unpublished style change (same for the body classes - e. g. ttu-book-body-wrapper)